src/ess/powder/grouping.py

```diff
@@ -12,23 +14,42 @@
 )
 
 
+def _drop_grouping_and_bin(
+    data: sc.DataArray, *, dims_to_reduce: tuple[str, ...] | None = None, edges: dict
+) -> sc.DataArray:
+    all_pixels = data if dims_to_reduce == () else data.bins.concat(dims_to_reduce)
+    # all_pixels may just have a single bin now, which currently yields
+    # inferior performance when binning (no/bad multi-threading?).
+    # We operate on the content buffer for better multi-threaded performance.
+    if all_pixels.ndim == 0:
+        content = all_pixels.bins.constituents['data']
```

Comment:
Is it safer to use this? Because it excludes constituents that are out of range?
```suggestion
        content = all_pixels.value
```

Reply:
Hmm, good point that looks simpler, I will try.